src/go/intelligence/utils/utils.go

```diff
@@ -54,7 +54,7 @@ var SuppressedAlertsDataReadWriteLock sync.RWMutex
 
 //DashboardsCache - a cache for storing dashboards and Expiration time for updating the cache
 type DashboardsCache struct {
-	Dashboards models.AllDashboardsFetched
+	Dashboards map[float64]models.AllDashboardsFetched
```

Comment:
Why map keys are float64? As far as I see from the code those are simple unsigned ints. Please double check and fix appropriately.

Reply:
This is because Grafana dashboard search API returns the value in float64. Here you can see the model for the dashboard struct.

```
type AllDashboardsFetched struct {
	ID          float64  `json:"id"`          //ID for the dashboard
	UID         string   `json:"uid"`         //UID for the dashboard
	Title       string   `json:"title"`       //Title for the dashboard
	URI         string   `json:"uri"`         //URI of the dashboard
	URL         string   `json:"url"`         //URL of the dashboard
	Slug        string   `json:"slug"`        //Slug of the dashboard
	Type        string   `json:"type"`        //Type of the dashboard
	Tags        []string `json:"tags"`        //All tags for the dashboard	(eg. prod, jobs, cmsweb etc.)
	IsStarred   bool     `json:"isStarred"`   //if dashboard is starred
	FolderID    float64  `json:"folderId"`    //ID for the folder
	FolderUID   string   `json:"folderUid"`   //UID for the folder
	FolderTitle string   `json:"folderTitle"` //Title of the folder
	FolderURL   string   `json:"folderUrl"`   //URL of the folder
}
```

Reply:
ok, thanks for checking (I was expecting something like that).